uwblueprint/paramedics-react
# Display maps

[Notion ticket](https://www.notion.so/uwblueprintexecs/396370c5e09f4c28bc7f455a7e84c908?v=3751aa62a2594c74b7e0ecb363304c24&p=419c95b8cc0d4a8ab4b2783744dfb16c)

## Changes
- Created Google Maps page
    - Displays location pins associated with a particular event
    - Built info window component to display pin information when pin is clicked
    - Added green location pin that indicates current location (enable this when prompted)
    - Integrated satellite view
- Added maps navigation to hamburger menu

## Screenshots
Pins displayed
<img width="684" alt="Screen Shot 2020-10-22 at 3 48 49 PM" src="https://user-images.githubusercontent.com/21224282/96934931-563ade00-1480-11eb-92fb-5bf5f19f1459.png">

Info window
<img width="683" alt="Screen Shot 2020-10-22 at 3 54 07 PM" src="https://user-images.githubusercontent.com/21224282/96934938-5804a180-1480-11eb-92b2-7d1403616196.png">

Hamburger menu
<img width="685" alt="Screen Shot 2020-10-22 at 4 04 51 PM" src="https://user-images.githubusercontent.com/21224282/96934942-59ce6500-1480-11eb-8c74-006044d81c80.png">
## Testing
Note that pins are supposed to represent the location of CCPs. For testing purposes, the pins displayed in this PR are random pins not associated with any CCP or event.

Before testing:
1. Update your backend: I have added new seeders to put pins in the Waterloo region. This makes testing much easier. Once the backend is updated, you can navigate to the 'Spooky Scary Event' -> click the hamburger menu -> navigate to the maps.
2. Add the Google Map API Key in `paramedics-react.env` which you can get [here](https://www.notion.so/uwblueprintexecs/6a0ad062ce504acc90fa25cb4d2c731f?v=4844a823da724b7ebf2c442a1ee1279f&p=e9959258451a48bba706904335794f57)

Note that this was styled based on the iPad Pro dimensions using the Chrome inspection (I need to define an absolute size of the map frame). Styling is still in progress and depends on the iPads that the paramedics are using.

Here are some tests to run
1. Ensure that all location pins are displayed
2. Ensure that the info window opened when a pin is clicked displays the correct information
3. Ensure that regular maps functionality is present (eg. zoom, current location, satellite vs. road map)

In my experience, the UI to test the changes on the Chrome iPad settings aren't great. If you do have an iPad, open up a tunnel and test it out.

## Checklist
>[Code review doc for reference](https://www.notion.so/uwblueprintexecs/Code-Review-a21fc85b00394f488e92d9d605f6b2bc)

before opening PR

filling out PR

after opening PR
